src/Stratis.PoAChainD/Program.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using NBitcoin;
+using Stratis.Bitcoin;
+using Stratis.Bitcoin.Builder;
+using Stratis.Bitcoin.Configuration;
+using Stratis.Bitcoin.Features.Api;
+using Stratis.Bitcoin.Features.Apps;
+using Stratis.Bitcoin.Features.BlockStore;
+using Stratis.Bitcoin.Features.MemoryPool;
+using Stratis.Bitcoin.Features.PoA;
+using Stratis.Bitcoin.Features.RPC;
+using Stratis.Bitcoin.Features.Wallet;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.PoAChainD
+{
+    public class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            try
+            {
+                Network network = new PoANetwork();
+                var nodeSettings = new NodeSettings(args: args, network: network);
+
+                bool keyGenerationRequired = nodeSettings.ConfigReader.GetOrDefault("generateKeyPair", false);
+                if (keyGenerationRequired)
+                {
+                    GenerateFederationKey(nodeSettings);
+                    return;
+                }
+
+                IFullNode node = new FullNodeBuilder()
+                    .UseNodeSettings(nodeSettings)
+                    .UseBlockStore()
+                    .UsePoAConsensus()
+                    .UseMempool()
+                    .UseWallet()
+                    .UseApi()
+                    .UseApps()
+                    .AddRPC()
+                    .Build();
+
+                if (node != null)
+                {
+                    await node.RunAsync();
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("There was a problem running the node. Details: '{0}'", ex.ToString());
+            }
+        }
+
+        private static void GenerateFederationKey(NodeSettings nodeSettings)
+        {
+            var tool = new KeyTool();
+            Key key = tool.GeneratePrivateKey();
```

Comment:
Keys will be generated outside of the deamon so they will just be provided in config

Reply:
Ok for now but when we release sidechains this needs to be a separate tool

Reply:
hmm actually maybe not, I get the argument this deamon is a tool and part of its functionality is to create a members mining key.